modules/nixos/fcitx5.nix

```
@@ -0,0 +1,31 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  inherit (config.catppuccin) sources;
+  cfg = config.i18n.inputMethod.fcitx5.catppuccin;
+  enable =
+    cfg.enable
+    && (
+      config.i18n.inputMethod ? enable
+      && config.i18n.inputMethod.enable
+      && config.i18n.inputMethod.type == "fcitx5"
+      || (!config.i18n.inputMethod ? enable && config.i18n.inputMethod.enabled == "fcitx5")
```

I'm pretty sure this can be simplified to the following. 
```suggestion
      config.i18n.inputMethod.enable or true
      && config.i18n.inputMethod.type or config.i18n.inputMethod.enabled == "fcitx5"
```
The first condition is the check for the newer 24.11 option, but should be ok to fall back on it being true as the next option will check the main `enabled` option from 24.05

---

Right, they should be identical. clever

---

Thank you very much!